modules/packages/LinearAlgebra.chpl

```diff
@@ -605,6 +604,60 @@ private inline proc _raw(D: domain(1), i) {
 // Matrix Structure
 //
 
+
+/*
+   Return a Vector containing the diagonal elements of ``A`` if the argument ``A`` is of rank 2.
+   Return a diagonal Matrix whose diagonal contains elements of ``A`` if argument ``A`` is of rank 1.
+ */
+proc diag(A: [?Adom] ?eltType){
+  if(Adom.rank == 2) then{
+    if(!isSquare(A)) then halt("diag supports only square matrices");
+    return _diag_vec(A);  
+  }
+  else if(Adom.rank == 1) then {
+    return _diag_mat(A);
+  }
+  else{
+    compilerError("A must have rank 2 or less");
+  }
+
+}
+
+/*
+  Return a Vector containing the off diagonal elements of ``A``
+*/
+proc offdiag(A:[?Adom] ?eltType){
```

Comment:
We weren't particularly clear when we wrote "off-diagonal", but what we meant was ``A[i,i+k]`` where ``k`` would be an input. A good model for what we were thinking of here is the equivalent numpy function --
  https://docs.scipy.org/doc/numpy/reference/generated/numpy.diagonal.html
(where you could ignore the axis pieces).

What you are using here is probably better referred to as the "skew diagonal", and it would be nice to have a function for that (although I can't immediately think of a use for it).